### PR TITLE
Add availableSupplies method to Board

### PR DESCRIPTION
**Pull request by tpsatish95**

### CODE REVIEW

File: src/io/sudhir/switchboard/boards/Board.java [hunk 1]
```diff
@@ -19,6 +20,8 @@ static Board create(Collection<? extends Supply> supplies, Collection<? extends
 
   boolean canProceed();
 
+  ImmutableSet<Supply> availableSupplies();
```
Comment (sudhirj) [inline src/io/sudhir/switchboard/boards/Board.java]: Switch to Set, don't expose Immutables.

**Review by sudhirj — CHANGES_REQUESTED** [review]

File: src/io/sudhir/switchboard/boards/Board.java (hunk 1 above)
Comment (tpsatish95) [reply src/io/sudhir/switchboard/boards/Board.java]: Shall I switch it to a `Stream<Supply>`?

Comment (sudhirj) [reply src/io/sudhir/switchboard/boards/Board.java]: No, that's too much drama. Set is fine.

Comment (sudhirj) [reply src/io/sudhir/switchboard/boards/Board.java]: Use a Stream for objects are likely to be lazily computed.

**Review by sudhirj — APPROVED** [review]

**Pull request merged**